### DISCUSSION TITLE
ESP32: Can't Update to 13.x.x due To Flash Size

### DISCUSSION THREAD

**Discussion by AlexKusnezov** [arendst/Tasmota#18983] [Q&A] [answered]
Hey guys, I can't update my ESP32, everytime I try I get
```
Program flash size is larger than real flash size
```

<img width="552" alt="Screenshot 2023-06-27 at 16 37 20" src="https://github.com/arendst/Tasmota/assets/33514406/06895cd8-2340-4344-98a0-d3a1810a2ec4">

Here's chip info:
```
ESP Chip Id | 401960 (ESP32-D0WDQ6 rev.1)
-- | --
0x1640D8 (DOUT)
4096 KB
4096 KB
1525 KB
1600 KB
147.4 KB (frag. 18%)
 
1600 KB
1600 KB (used 95%)
60 KB
```

FYI:
I also have this one, which also wasn't updating. But this one I managed to update by using tasmota32-bluetooth. The other one above didn't work with this workaround.
```
ESP Chip Id | 13778000 (ESP32-D0WD-V3 rev.3)
-- | --
0x16405E (DOUT)
4096 KB
4096 KB
1647 KB
1792 KB
81.9 KB (frag. 6%)
 
1792 KB (used 91%)
1792 KB
60 KB
```

Any idea how to update?

**Comment by sfromis** [comment]
You did neither tell what specific Tasmota binary you're already on, nor what exactly you do to update (including what binary URL/file), nor what your device is, nor whether you have the device in safeboot partitioning scheme.

**Reply by AlexKusnezov** [reply]
I'm on tasmota32.bin (12.5.0)
I always just downloaded the tsamota32.bin and udated my device manually, which is not working since 12.5.0.
Binary is http://ota.tasmota.com/tasmota32/release/tasmota32.bin Device is ESP32-DevkitC (NodeMCU)
I included chip specs above.
>whether you have the device in safeboot partitioning scheme.

Do you inow where I can see that? Update worked always flawlessly until now! (13.0.0)

**Reply by sfromis** [reply, marked as answer]
With safeboot (which you likely do not have), the information page ends with settings like this:
![image](https://github.com/arendst/Tasmota/assets/47082390/f5cfffdc-19f0-4375-a263-ea803b9f7ba1)

With Tasmota 13.0, support for Matter is now included by default, which means that the Tasmota binaries now exceed the size of the two equal-sized partitions from before the [safeboot](https://tasmota.github.io/docs/Safeboot/) partitioning scheme. Thus you need to first upgrade your setup to the safeboot layout. You can do this by [running the Partition Wizard](https://tasmota.github.io/docs/Tasmota-Application/#partition-management) tool, without reinstalling from scratch.

If you instead do a fresh install of Tasmota, using the recommended [Tasmota Web Installer](https://tasmota.github.io/install/) tool, you'll also get a setup using the safeboot partitioning. That works by installing `tasmota32.factory.bin` at offset 0. If you do this, any existing Tasmota install is overwritten.

**Reply by AlexKusnezov** [reply]
cool, if I run the partition wizard will the config be wiped out/ can I restore it afterwards?

**Reply by sfromis** [reply]
Config will not be wiped if you use the Partition Wizard, that is only if you reinstall from scratch. Better have backup of anything you have in the file system.

**Reply by AlexKusnezov** [reply]
thanks, that's exactly waht I did and it worked right away!

**Reply by grahamsingle** [reply]
I've have 12.5.0 and when I try to upload "Partition Wizard.tapp" from "Manage File System" nothing happens. The file doesn't appear in the file system after I choose the file press "Start Upload". 

I get the following from info:
Partition app0         1856 KB
Partition app1*       1856 KB (used 82%)
Partition fs	       320 KB

Maybe there's not enough room for the Wizard, but no error is displayed and the file files are on the file system as before. Does the * suggest I'm running from app1?

Any suggestions?

**Reply by grahamsingle** [reply]
Figured it out. Managed to get Partition_Wizard.tapp on the device using urlfetch to a URL on a webserver on my network, then rebooted and run Partition Wizard. Hope this helps some one in the same predicament.

**Reply by Sysselix** [reply]
That worked for me: [running the Partition Wizard] and expanding FS. The Device has upgraded successfully to the lastest Tasmota Version! 

Before:
![2024-01-29 00_15_37-Tasmota Kühl Override - Information – Mozilla Firefox](https://github.com/arendst/Tasmota/assets/136033764/8140220f-adec-4b0f-beef-31f447729107)

Finished:
![2024-01-29 00_28_03-Tasmota Kühl Override - Information – Mozilla Firefox](https://github.com/arendst/Tasmota/assets/136033764/76b26361-ce44-4aa2-bcd3-49af2b28ff7f)

Reached: [Tasmota 13.3.0(solo1) by Theo Arends]

Thanks!!!

**Reply by ssiillvvyyaannaa** [reply]
Good night
Any tutorial to run Partition Wizard on Tasmota with ESP32.
Thanks

**Reply by barbudor** [reply]
It may depend which version you currently have 
But I would say that the idea is to upgrade a recent binary of a small variant that will fit into your flash. A binary around 1.2M would be ok such as https://ota.tasmota.com/tasmota32/tasmota32-ir.bin
Once you have upgraded, you will find the Partition Wizard in the "Tools" menu
Just click and follow the instructions
After repartitionning, you can upgrade to any tasmota32 variant you like

**Reply by ssiillvvyyaannaa** [reply]
barbudor, Thank you so much
Program Version: 12.5.0 (tasmota32)
¿ I would not want to lose the entities, the HomeAssistant automations would fail. ?
My old configuration is lost..???
Thank you

**Reply by ssiillvvyyaannaa** [reply]
Good night
Along this path, what am I doing wrong?
Thank you so much

![010](https://github.com/arendst/Tasmota/assets/57443558/39aef4b3-981e-40c6-9772-018d3bb5727b)
![020](https://github.com/arendst/Tasmota/assets/57443558/19d2aca8-f178-46a2-837d-154cbcff97f9)
![030](https://github.com/arendst/Tasmota/assets/57443558/d571a9c7-540b-4fb5-b17c-064ef4b64a8b)

**Restart**

![040](https://github.com/arendst/Tasmota/assets/57443558/495da8f9-4262-42f6-9b5d-9b33d6bfbebe)

**Reply by ssiillvvyyaannaa** [reply]
Solved the 1st part thanks to **s-hadinger:**
https://github.com/arendst/Tasmota/blob/development/tasmota/berry/modules/Partition_Wizard.tapp

![060](https://github.com/arendst/Tasmota/assets/57443558/9c8baeb0-ff37-4906-9e24-7ab2ba78c38e)
![050](https://github.com/arendst/Tasmota/assets/57443558/38bde712-f3b4-4f93-a5bd-7ba3ce264b06)

**Reply by Jason2866** [reply]
Your partition file system is (too) small. You did the initial flashing probably with Tools not made to be used with Tasmota.
Resize the Filesystem partition to 320k

**Reply by ssiillvvyyaannaa** [reply]
Thank you Jason2866
In this thread there are several links to download the "Partition_Wizard.tapp", just download an INVALID one.
The valid one was entered and the first part was ready, in "CONSOLES" the "Partition Wizard" now appears (35680 the good one; 24408 the bad one)

Solved the 1st part thanks to s-hadinger:
https://github.com/arendst/Tasmota/blob/development/tasmota/berry/modules/Partition_Wizard.tapp

**Reply by ssiillvvyyaannaa** [reply]
Everything is finished but on the 5 devices that I have partitioned, the partitions do not fit me like they do, I hope I do not have problems in the future.
Thank you all.

![090](https://github.com/arendst/Tasmota/assets/57443558/6744c027-6e2a-428e-8fb6-0a3fd1002e42)

**Comment by ArtjomE** [comment]
I had exactly the same issue with Shelly Plus 1PM (ESP32-U4WDH-S rev.3) **"Program flash size is larger than real flash size"** by upgrade from 12.5.0 to 13.0.0

```
Program Version 12.5.0(solo1)single-core
Build Date & Time 2023-04-17T08:07:06
Core/SDK Version | 2_0_7/4.4.4.20230218
```

I did a firmware upgrade with _tasmota32solo1-safeboot.bin_ and now I have although new version 13.0.0 but just in SAFEBOOT

```
Program Version 13.0.0(safeboot)single-core
Build Date & Time 2023-06-26T10:42:56
Core/SDK Version 2_0_10/4.4.5.230614
```
and the problem with the firmware update with the _tasmota32solo1.bin_ has continued and show me the same message as before **"Program flash size is larger than real flash size "**.

by upload the _tasmota32solo1.factory.bin_ i got the message **"Upload Failed" "Invalid file signature"**

**Reply by sfromis** [reply]
To upgrade to version 13 you needed to do the repartitioning I already suggested. Not sure why you decided to try updating to the safeboot binary, instead of following the instructions, but this just made your situation worse, as the safeboot binary is not suitable for running the device.

You got yourself into some "additional trouble", but if you want to try to salvage it without reinstalling, you could try the command `Restart 3` to get to the other regular partition, and then use the Partition Wizard to change partition scheme. Not 100% sure if this will work, but I think that the chances are not too bad. You may want to consider the risk of time spent on repairs being longer than reinstalling.

**Reply by ArtjomE** [reply]
_this is just what happens when you try everything first and then google for solutions_ 🙈

the command  `Restart 3` does not work in console `MQT: stat/TasmotaCarportOutside/RESULT = {"Command":"Error"}`

how can I reinstall it, I do not care about the data ?

**Reply by sfromis** [reply]
Reinstall can be done via the externally available (no need for removing the cover) pins shown here:
https://templates.blakadder.com/shelly_plus_1PM.html

**Reply by ArtjomE** [reply]
is there no way to do this via OTA ?
ota with `http://ota.tasmota.com/tasmota32/shelly/mgos32-to-tasmota32-Plus1PM.zip` does not work

PS:
i got my shelly back to life with flash version 12.5.0 of file `tasmota32solo1.bin`

**Reply by sfromis** [reply]
Well, OTA is the normal way of upgrading, but when you screwed it up, any repair attempt after that becomes somewhat experimental, meaning that I'm not saying what will not work, or what will work. And experimental repair can be time-consuming, especially if trying to guide someone over the Internet, while wired flashing is a quick and easy process, when you have the hardware setup for it, and use the Tasmota Web Installer.

Anyway, great if you got it to work.

**Comment by AlexKusnezov** [comment]
> _this is just what happens when you try everything first and then google for solutions_ 🙈
> 
> the command `Restart 3` does not work in console `MQT: stat/TasmotaCarportOutside/RESULT = {"Command":"Error"}`
> 
> how can I reinstall it, I do not care about the data ?

well, the thing is. What the most of users do is just hit the firmware update and expect it to work like it has woked the last 10 times before. In the changelog I couldn't see anything like "you should convert your esp32's partition with partition wizard in order to be able to update to v13" 
Maybe I'm just dumb but all I've red was
```
ESP32 Tasmota SafeBoot with changed partition scheme allowing larger binaries
```
and thought, yeah cool so what?

Imho such guides, like usage of the partition wizard or at least to mention it should be included in the chnagelog.

Also thanks to @sfromis who pointed us, ignorant update button pushers, into the right direction

**Reply by arendst** [reply]
The complete releasenotes just tells you that under the changed heading

**Reply by sfromis** [reply]
> ESP32 Binaries size increase of 300k due to Matter support may need Partition Wizard

Arguably, it could be called a "breaking" change to some users, but lots of users already on the safeboot partitioning scheme are not affected.

**Reply by AlexKusnezov** [reply]
@arendst really? I'm sorry, but I didn't noticed any **Chnaged** section under 13.0.0 only **Added**, I'm looking here: https://github.com/arendst/Tasmota/releases

Can you please point me to the right entry?

**Reply by arendst** [reply]
How about this https://github.com/arendst/Tasmota/releases/tag/v13.0.0

**Reply by AlexKusnezov** [reply]
that's perfect! Thanks you so much, never thought that descriptions from tags differ from releases.

**Reply by meyergru** [reply]
Since I fell for the same mistake, I can only say that this could have stood out a little more aggressively like under "breaking changes". There are devices that cannot be easily re-flashed via serial without relocating them first.

You remember the Monty Python clip: https://www.youtube.com/watch?v=1oyCXhiwulk ? "It should bear a large red label "Warning: lark's vomit!'"

**Comment by blacknell** [comment]
I found the upgrade to 13.0 from 12.4 wouldn't work even with safeboot 12.1 already installed. I think a more recent version of 12.x safeboot might be needed?

I found these in the logs
```
Jul  3 10:37:39 tasmota-BAE078-0120 ESP-Project tasmota - temphumid_loft Version 12.1.0(safeboot)-2_0_4_1(2022-08-18T10:33:31)
Jul  3 10:37:39 tasmota-BAE078-0120 ESP-WARNING This version does not support persistent settings
```
I executed `Upgrade 2` from normal boot mode which updated the boot partition to 13.0 and then `Upgrade 1` worked as normal.

I couldnt see anything that defined what is the oldest version of safeboot need for this to work

**Reply by arendst** [reply]
Yes. If safeboot fails to start the upgrade it's likely due to an old version (<12.4). Later versions use a different way of signalling an upgrade.

Your solution is the correct one.

**Reply by sfromis** [reply]
I've gotten into the habit of occasionally upgrading safeboot to current version, indeed using `Upgrade 2`. If wanting to switch out of safeboot (or to safeboot), the command `Restart 3` can be useful, instead of doing some upgrade just to switch partitions.

**Reply by Noschvie** [reply]
How can the version of the safeboot partition be checked? Thanks

**Reply by sfromis** [reply]
By using the restart command I just described in the previous post here. That allows you to switch partition, and see what you have, without performing an upgrade. Arguably, having reason to check is also having reason to just upgrade :grin:

**Comment by ReelHiJacker** [comment]
I've got several Shelly Plus 1 with 12.5.0. installed. I like to upgrade them, which doesn't work like you mention above. I'v tried the Partion Wizard Solution but i'v got this error in console after upload the file and restart:

TAP: Loaded Tasmota App 'Partition_Wizard.tapp'
BRY: failed to run compiled code 'io_error' - inconsistent number of builtin objects.

Partition Manager works, but not the Wizard... now i'm stuck... can you help me?
Tried it on different Shelly Plus 1 ... everytime the same error...

**Reply by sfromis** [reply]
There was a recent change to Partition_Wizard to fix another situation, I'd suggest trying with the next older build:
https://github.com/arendst/Tasmota/raw/eb3de9c73a23e5559123dd85df70f844aaa46f33/tasmota/berry/modules/Partition_Wizard.tapp

**Reply by ReelHiJacker** [reply]
This works... Thank you! :-)

**Reply by sfromis** [reply]
Looks like Partition Wizard needs another fix, @s-hadinger

**Reply by s-hadinger** [reply]
I need to fix some stuff. Please be patient

**Reply by s-hadinger** [reply]
This is now fixed. Please use the latest from the repo:
https://github.com/arendst/Tasmota/blob/development/tasmota/berry/modules/Partition_Wizard.tapp

**Reply by cschoell** [reply]
I get a "UPL: error (network_error) GET returned 302" whenever I try to use the Partition Wizard - (it looks like the safeboot url is redirected from http to https) 
... turns out I blocked the tasmota device from internet access via my router - it's working fine.

**Comment by foobar26** [comment]
Hi,
I wanted to upgrade my Shelly Plus 2PM from 12.5.0 to 13 via the Partition manager approach.
The first steps went fine, it seems.
But now I'm stuck in Safeboot mode.
I tried to switch witch Restart 3, but still end up in Safeboot mode.
Restart 1 same thing -> safeboot mode.
I also tried to upgrade FW, same result.
Information page shows at the end:

```
Partition safeboot*	832 KB (used 94%)
Partition app0	2688 KB (used 69%)
Partition fs	448 KB

```

I'm not sure why the device doesn't want to switch to the app0 partition. Any way to force it?

**Reply by sfromis** [reply]
Which binary did you try to upgrade to? If not `tasmota32solo1.bin`, try doing that.

**Reply by foobar26** [reply]
tried that. again ending up in safeboot

**Reply by foobar26** [reply]
used also upgrade 1 or 2. no difference.

**Reply by foobar26** [reply]
I ended up flashing the device via serial :-\

**Reply by fir3drag0n** [reply]
Unfortunately the problem still exists

**Comment by Jason2866** [comment]
@foobar26 The reason for the Shelly failing to convert successfully was an updated more secure Shelly firmware. It has nothing to do with the Safeboot convert in general. Safeboot convert works well.
The actual Shelly firmwares uses a locked bootloader. When this bootloader is not replaced the device gets stuck on the firmware which is at address `0x10000`. No way out. Just serial flashing can solve.
Since 19.Sept 2023 we have an updated [Shelly Convert to Tasmota ](https://github.com/tasmota/mgos32-to-tasmota32)online.
Shelly converting is a cat and mice game. Let's see how long it will last this time.

**Reply by foobar26** [reply]
@Jason2866 thx for your explanation and adding the extra warning to the migration process for mgos32-to-tasmota32!

**Comment by Jason2866** [comment]
@peepshow-21 Wrong Berry App. You have Partition Manager running. For converting PartitionWizard is needed.

**Reply by peepshow-21** [reply]
sorry, realised that, and deleted my silly question before i saw your reply!! :)
thanks you

**Comment by oywino** [comment]
Maybe not exactly the same problem - but very similar;
I have many SonOff switches and bridges, all flashed with Tasmota 13.2.0 but this seems to be the end of it ?
Going forward I no longer can flash even the Tasmota-minimal.bin as there just isn't enough memory!
So what then?
Are we stuck at 13.2.0 ?

**Reply by arendst** [reply]
Totally offtopic. This is about ESP32 based evices.

Yours are ESP8266 based and there is no issue upgrading these to latest versions other than bad power supplies not able to support latest features

**Reply by sfromis** [reply]
For ESP8266, as indicated on https://ota.tasmota.com/tasmota/release/ the `.bin` files are **only** "initial firmware" for wired flashing, while upgrading should *always* be with the `.bin.gz` files.

When doing upgrades OTA, do not download first, and do not use `-minimal.bin.gz` as this will be automatically handled by the upgrade process.

**Reply by oywino** [reply]
OMG - Thank you so much for that information.  I've been using Tasmotized devices for years, and not once have I been able to successfully complete an OTA upgrade. So I guess I (wrongly) concluded long ago that OTA just doesn't work.
Based on your reply above, I now decided to give it another go - since my year-long faithful manual file based method had come to a halt. In fact, I didn't have much choice.
It's a nerve wrecking experience.  I first checked the URL and then closed my eyes and pressed the button. Not much feedback indicating what was going on or if anything at all, but after a while it came back online with the minimal FW and requested another upgrade. So far, so good. <Sigh> 
Then pressing upgrade again - nothing happens, waiting--- and it was gone 404.   Ok that's it I thought. another bricked device. I went for a cup of coffee. And guess what! Upon my return som 15 minutes later, it had miraculously recovered all by itself and now with Tasmota 13.3.0   Joy to the people! 😄❤️👍
Thank you so much. 
But two questions remain in my mind:
1. How come I haven't found that vital bit of information anywhere before; 
`
When doing upgrades OTA, do not download first, and do not use -minimal.bin.gz as this will be automatically handled by the upgrade process.
`
Honestly - it isn't very obvious that the `minimal.bin.gz` has been provided so we should have to use it - eh? I guess it serves some other mysterious purpose beyond being used.

2.  I wish the upgrade process would provide some sort of visual on screen progress feedback. I almost unplugged the device thinking it was bricked. Thank God for the coffee break 😊

**Reply by Jason2866** [reply]
Point 1) Documented here https://tasmota.github.io/docs/Upgrading/
Point 2) Not possible during upgrade

**Comment by forestial** [comment]
I'm looking for the correct upgrade path for a SwitchBot device that I have.  This uses an ESP32-C3.  I have been able to get Tasmota 11.1.0 on it using OTA and the 'switchbota' tool, but it will not upgrade past that; I presume because of the partition change.  The case of the device is 'welded' and I would really like to avoid cutting it open, so I am looking for a way to do this OTA.

From reading the replies above I think I need to 
1. Load the Partition Manager TAPP file
2. Power cycle the device.
.... I'm not clear what I would do next.  Do I need to issue one of the 'restart' commands?  Will tasmota still be functional at this point so that I can access its web page interface?  Can I just attempt OTA upgrade again, and will it now install the latest v13.x?

**Reply by barbudor** [reply]
You should have got a new menu "Partition Wizard' to appear in the "Consoles" menu

I just tried and actually Partitition Wizard doesn't run on a Tasmota 11.1.0 (missing a component)

So I upgraded to the latest tasmota32c3-safeboot from https://ota.tasmota.com/tasmota32/tasmota32c3-safeboot.bin
which worked fine
Then upgraded to the latest tasmota from https://ota.tasmota.com/tasmota32/tasmota32c3.bin
You can either download the binary and upgrade by upload, or durectly use the OTA upgrade

Now you should have a "Tools" menu with "Partition Wizard" which would allow you to start the migration
When you have started the migration, do not touch the board for at least 3 to 4 minutes
If it doesn't come back by itself, re-open the main page
If for some reason you are in Safeboot, just re-upgrade once again to full version 
And you should be done 
No more upgrade problems in the future

**Reply by forestial** [reply]
Thanks.  I did the first step (upgrade to tasmota32c3-safeboot.bin) successfully, but when I try the second step (upgrade to latest tasmota32c3.bin) directly with OTA, it just remains at 13.3.0.4(safeboot).  In the console, I see:
13:09:58.132 MQT: house/SBTasm08/stat/RESULT = {"Upgrade":"Version 13.3.0.4 from https://ota.tasmota.com/tasmota32/tasmota32c3.bin"}
13:10:15.699 MQT: house/SBTasm08/stat/UPGRADE = {"Upgrade":"Failed "}

I also tried (as suggested) to download tasmota32c3.bin locally and upgrade from there, but that gives an error:
Upload Failed

Program flash size is larger than real flash size

So I don't have the 'Tools' menu, and I can't proceed with the Partition Wizard.

**Reply by barbudor** [reply]
Can you post a screen shot of the bottom of the information page with the size of flash, programs, etc...?

**Reply by forestial** [reply]
![image](https://github.com/arendst/Tasmota/assets/9161932/9b6b0ccd-ec0a-4049-b129-13a8ae612ffd)

**Reply by barbudor** [reply]
Ahhhh 1856K is not enough, tasmota32c3 is 1878k
Let me build a shrinked version of tasmota23c3 that would fit

**Reply by Jason2866** [reply]
This one should be small enough https://github.com/Jason2866/Tasmota-specials/releases/download/2024-01-30.d.95b03592dc0b753dfcfa30c1cc17d8e1f2a547f9/tasmota32c3_2M.bin

**Reply by barbudor** [reply]
AH thanks @Jason2866

**Reply by barbudor** [reply]
@forestial So upgrade to the one pointed by Jason
This will allow you to run the PartitionWizard (you may need to re-upload it)
Once you have migrated to the new safeboot structure, you can then upgrade to the standard tasmota

**Reply by forestial** [reply]
Thanks guys!  I used Jason's '2M' version, then ran the Partition Wizard, then upgraded to the regular tasmota32.bin.  (Had to repeat that a couple of times, for some reason it stayed with the safeboot initially).  Now I see
![image](https://github.com/arendst/Tasmota/assets/9161932/6a985e61-0062-46fc-8019-c257b184d742)

**Reply by barbudor** [reply]
Congrats

**Comment by oywino** [comment]
For ESP based devices, as indicated on https://ota.tasmota.com/tasmota/release/ the .bin files are only "initial firmware" for wired flashing, while upgrading should always be done using the .bin.gz files.
When doing OTA upgrades, do not download the FW files, and do not use -minimal.bin.gz as this will be automatically handled by the OTA upgrade process.

**Reply by barbudor** [reply]
You are out of context. This thread is about ESP32

**Comment by DcFamas** [comment]
Hey guys,
I also want to update. I'm to stupid to follow those steps. Could you maybe assist me?

I already updated to 12.5.0, form 10, on my Yeelight Meteorite. I also uploaded the mentioned Partition_Wizard.tapp. But I still cannot see the entry under Consoles. I'm also not able to delete the Partition_Wizard.tapp anymore.

![grafik](https://github.com/user-attachments/assets/a581fdd1-0d28-4e21-9b91-7e32a71023f3)

![grafik](https://github.com/user-attachments/assets/9b9304b3-4c02-49a1-883e-1739c927d443)

I want to test matter, thats the reason why updating. :)
Thank you guys!

**Reply by sfromis** [reply]
As `.tapp` files are loaded at restart, that's what it take for the Partition Wizard to appear.

**Reply by DcFamas** [reply]
Hey @sfromis thanks for the quick response! :) Unfortunately I already restartet the device several times. Including to power cycle it completly.

**Reply by whatever86-moneytrain** [reply]
Hey @DcFamas , I had the same problem.
Download this [PartitionWizard](https://github.com/arendst/Tasmota/raw/eb3de9c73a23e5559123dd85df70f844aaa46f33/tasmota/berry/modules/Partition_Wizard.tapp)
rename it to Partition_Wizard_old.tapp and upload it to your fs in ManageFileSystem. After a reboot you can delete the Partition_Wizard.tapp and Partition Wazard should be appear under Console.
In my case that doesn't helped, because the Partition_Wizard showed only the Partitions and no "Migration" button.
Than I upgraded with [tasmota32-ir](https://ota.tasmota.com/tasmota32/release-13.4.0/tasmota32-ir.bin) to version 13.4.0, because witch the current "tasmota32-ir" release you can't do the migration. After that you can do the migration with the Partition_Wizard and it will also upgrade to the current release, to do that your device needs a internet connection. But be carefull "tasmota32-ir" is only for the ESP32 with a dualcore.

**Reply by DcFamas** [reply]
Works! Thank you! :)

**Reply by Matten-Matten** [reply]
@whatever86-moneytrain  Funktioniert besten Dank!

**Comment by Dyrke** [comment]
I have the problem that everytime I use the web update og my D0WD/D0WDQ6 types of esp32 it always goes into safeboot. Now I have been advised to use the tasmota32-lite.bin first. But the only version I can find is a version 9.3.1.
Is it okay to use that and the update to ex 15.1,0

**Reply by sfromis** [reply]
Whoever told you to use `tasmota32-lite-bin` simply did not know how it works. Was it some AI LLM thing? Not only wrong, but not even a plausible direction.

And it would not make sense to dig up old crusty versions, as they would not apply to current upgrading issues.
